rust-symcrypt/src/gcm.rs

```diff
@@ -4,97 +4,119 @@ use crate::block_ciphers::*;
 use crate::errors::SymCryptError;
 use core::ffi::c_void;
 use std::mem;
+use std::pin::Pin;
 use std::ptr;
 use std::vec;
 use symcrypt_sys;
 
 pub struct GcmState {
-    expanded_key: symcrypt_sys::SYMCRYPT_GCM_EXPANDED_KEY,
+    inner: Pin<Box<GcmStateInner>>,
```

Comment:
Doesn't need to be changed for this PR, but I think you'll have to go back and update the other Rust structures you've created to use this same pattern (e.g. for hash, HMAC). Those SymCrypt structures are also designed to not be `memcpy`'d, and also have magic fields. I'm not sure why you didn't hit errors before with those.

Reply:
Have it tracked as an issue